SWIPs/swip-bzz-handshake.md

```diff
@@ -0,0 +1,69 @@
+---
+SWIP: <to be assigned>
+title: Swarm node implementer spec - 
+author: Louis Holbrook @nolash <dev@holbrook.no>
+status: Draft
+type: Track Specs
+created: 2019-08-08
+---
+
+## Abstract
+
+This SWIP is a part of a general specification of a Swarm node. The SWIP system is used for convenience only.
+
+## Motivation
+
+Enable other implementations of the swarm node.
+
+## Backwards Compatibility
+
+N/A
+
+## Test Cases
+
+N/A
+
+## Implementation
+
+N/A
+
+## Copyright
+
+Copyright and related rights waived via CC0
+
+## Specification
+
+### bzz protocol
+
+#### handshake
+
+The bzz handshake is the first message to be exchanged between peers
+after the p2p transport protocol has been successfully negotiated. After
+the handshake each peer should remember the following data about the
+other:
+
+  - `Swarm Overlay Address` of the peer [\[address\]](#address)
+
+  - What `Capabilities` the peer has advertised
+    [\[capabilities\]](#capabilities)
+
+##### message
+
+The handshake message consists of four data fields:
```

Comment:
how are they encoded? might be useful to more clearly point out dependencies/requirements to do this handshake

Reply:
I'm not sure how to define this here, as it would depend on the serialization, and the main spec section is serialization neutral.